module/x/gravity/keeper/ethereum_event_vote.go

```diff
@@ -50,7 +50,11 @@ func (k Keeper) recordEventVote(
 	// Add the validator's vote to this EthereumEventVoteRecord
 	eventVoteRecord.Votes = append(eventVoteRecord.Votes, val.String())
 
-	k.setEthereumEventVoteRecord(ctx, event.GetEventNonce(), event.Hash(), eventVoteRecord)
+	// ignore "old" event votes but still record the nonce for the submitting validator
+	if event.GetEventNonce() > k.GetLastObservedEventNonce(ctx) {
+		k.setEthereumEventVoteRecord(ctx, event.GetEventNonce(), event.Hash(), eventVoteRecord)
+	}
+
```

Comment:
Is there potential for multiple events with nonces greater than the last observed to be called out of order and thus end up with `setLastEventNonceByValidator` setting it incorrectly?

Reply:
if so this would already be an issue right?

Reply:
There are many things in this refactor that we are fixing that are already issues, curious if this one is as well.

Reply:
bumping this

Reply:
The check at the beginning of this method enforces the nonces being increments of 1 after the other; e.g. it won't record 3 on top of 1. Is that what you're asking?